storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -526,4 +524,13 @@ private void writeToFile() {
             lock.readLock().unlock();
         }
     }
+
+    private void writeToFileForTruncation() {
+        lock.readLock().lock();
+        try {
+            checkpoint.writeForTruncation(epochs.values());
```

Comment:
Could we add a comment on why it's important to hold the lock while writing the checkpoint file?